src/flamegraph/color.rs

```diff
@@ -0,0 +1,272 @@
+            "aqua" => Ok(Palette::Aqua),
+            "yellow" => Ok(Palette::Yellow),
+            "purple" => Ok(Palette::Purple),
+            "orange" => Ok(Palette::Orange),
+            unknown=> Err(format!("unknown color palette: {}", unknown))
+        }
+    }
+}
+
+fn namehash(name: &str) -> f32 {
+    let mut vector = 0f32;
+    let mut weight = 1f32;
+    let mut max = 1f32;
+    let mut modulo = 10;
+
+    let name = {
+        let looking_for_module_name = if name.starts_with("`") {
+            &name[1..]
+        } else {
+            name
+        };
+
+        if let Some(index) = looking_for_module_name.find("`") {
+            &looking_for_module_name[index + 1..]
+        } else {
+            name
+        }
+    };
+
+    for character in name.bytes() {
+        let i = (character % modulo) as f32;
+        vector += (i / ((modulo - 1) as f32)) * weight;
+        modulo += 1;
+        max += weight;
+        weight *= 0.70;
+
+        if modulo > 12 {
+            break
+        }
+    }
+
+    (1f32 - vector / max)
+}
+
+fn handle_java_palette(s: &str) -> Palette {
+    if s.ends_with("]") {
+        if let Some(ai) = s.rfind("_[") {
+            if s[ai..].len() == 4 {
+                let suffix_char = &s[ai+2..ai+3];
+                if suffix_char.starts_with("k") { return Palette::Orange }
+                if suffix_char.starts_with("i") { return Palette::Aqua }
+                if suffix_char.starts_with("j") { return Palette::Green }
+            }
+        }
+    }
+
+    let java_prefix = if s.starts_with("L") { &s[1..] } else { s };
+
+    if java_prefix.starts_with("java/") ||
+        java_prefix.starts_with("org/") ||
+        java_prefix.starts_with("com/") ||
+        java_prefix.starts_with("io/") ||
+        java_prefix.starts_with("sun/") {
+        Palette::Green
+    } else if s.contains("::") {
+        Palette::Yellow
+    } else {
+        Palette::Red
+    }
+}
+
+fn handle_perl_palette(s: &str) -> Palette {
+    if s.contains("::") {
+        Palette::Yellow
+    } else if s.contains("Perl") || s.contains(".pl") {
+        Palette::Green
+    } else if s.ends_with("_[k]") {
+        Palette::Orange
+    } else {
+        Palette::Red
+    }
+}
+
+fn handle_js_palette(s: &str) -> Palette {
+    if s.ends_with("_[j]") {
+        if s.contains("/") {
+            return Palette::Green
+        } else {
+            return Palette::Aqua
+        }
+    } else if s.contains("::") {
+        return Palette::Yellow
+    } else if let Some(ai) = s.find("/") {
+        if (&s[ai..]).contains(".js") {
+            return Palette::Green
+        }
+    }
+
+    if s.contains(":") {
+        Palette::Aqua
+    } else if s == " " {
+        Palette::Green
+    } else if s.contains("_[k]") {
+        Palette::Orange
+    } else {
+        Palette::Red
+    }
+}
+
+fn handle_wakeup_palette(_s: &str) -> Palette {
+    Palette::Aqua
+}
+
+fn handle_chain_palette(s: &str) -> Palette {
```

Comment:
I'd cut this since we don't handle `chain` correctly anyway.

Reply:
And remove the MultiPalette::Chain entry totally?
`chain` is not in the `possible_values` of the `colors` option, I think it is sufficient for now.

Reply:
Yeah, I would remove it totally so that we don't have dead code laying around :)

Reply:
[Done](https://github.com/jonhoo/inferno/pull/32/commits/2c486fe79d8cc76cf5cf43ba0ff5479cb0006293)